### PR TITLE
feat: support adjusting document width

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

closes https://github.com/AppFlowy-IO/AppFlowy/issues/6312

update:

Screenshot 2024-09-17 at 15 47 06

https://github.com/user-attachments/assets/49d4f951-12ce-4833-9dff-c7804a1bd0ae

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/pages/settings_workspace_view.dart [hunk 1]
```diff
@@ -1266,3 +1266,90 @@ class _SelectionColorValueWidget extends StatelessWidget {
...
+  });
+
+  final void Function(double) onPaddingChanged;
+
+  @override
+  State<_DocumentPaddingSlider> createState() => _DocumentPaddingSliderState();
+}
+
+class _DocumentPaddingSliderState extends State<_DocumentPaddingSlider> {
+  double width = 480;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/pages/settings_workspace_view.dart:1320]:
```suggestion
  late double width;
```

**Review by Xazin — APPROVED** [review]

**Comment by emmggi** [comment]
I like this. It'll be much easier to follow the content on wide screens.

Is the last step in the slider for unlimited width?

**Comment by Xazin** [comment]
> I like this. It'll be much easier to follow the content on wide screens.
> 
> Is the last step in the slider for unlimited width?
> 
> Also I'm not sure if it would make sense to put the slider in the menu with font size, but you could easily see the width of the document as you resize.

Added an image of the menu you speak of :+1:

image

I guess we could also add it there.

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/pages/settings_workspace_view.dart (hunk 1 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/pages/settings_workspace_view.dart]: I updated the max and min values and made them const values.

**Comment by emmggi** [comment]
> customize the document width for a specific page

Not sure how many users would benefit this but we'll see how it goes. :shrug:

**Pull request merged**